simtools/visualization/visualize.py

```diff
@@ -699,3 +703,347 @@ def plot_array(telescopes, rotate_angle=0, show_tel_label=False):
+    log_z: bool
+        if True, the intensity of the color bar is given in logarithmic scale.
+    """
+    return _kernel_plot_2D_photons(corsika_output_instance, "density", log_z=log_z)
+
+
+def plot_2D_direction(corsika_output_instance, log_z=True):
+    """
+    Plot the 2D histogram of the incoming direction of photons.
+
+    Parameters
+    ----------
+    corsika_output_instance: corsika.corsika_output.corsikaOutput
+        instance of corsika.corsika_output.corsikaOutput.
+    log_z: bool
+        if True, the intensity of the color bar is given in logarithmic scale.
+    """
+    return _kernel_plot_2D_photons(corsika_output_instance, "direction", log_z=log_z)
+
+
+def plot_2D_time_altitude(corsika_output_instance, log_z=True):
+    """
+    Plot the 2D histogram of the time and altitude where the photon was produced.
+
+    Parameters
+    ----------
+    corsika_output_instance: corsika.corsika_output.corsikaOutput
+        instance of corsika.corsika_output.corsikaOutput.
+    log_z: bool
+        if True, the intensity of the color bar is given in logarithmic scale.
+    """
+    return _kernel_plot_2D_photons(corsika_output_instance, "time_altitude", log_z=log_z)
+
+
+def plot_2D_num_photons_per_telescope(corsika_output_instance, log_z=True):
+    """
+    Plot the 2D histogram of the number of photons per event and per telescope.
+
+    Parameters
+    ----------
+    corsika_output_instance: corsika.corsika_output.corsikaOutput
+        instance of corsika.corsika_output.corsikaOutput.
+    log_z: bool
+        if True, the intensity of the color bar is given in logarithmic scale.
+    """
+    return _kernel_plot_2D_photons(
+        corsika_output_instance, "num_photons_per_telescope", log_z=log_z
+    )
+
+
+def _kernel_plot_1D_photons(corsika_output_instance, property_name, log_y=True):
+    """
+    Create the figure of a 1D plot. The parameter `name` indicate which plot. Choices are
+    "counts", "density", "direction".
+
+    Parameters
+    ----------
+    corsika_output_instance: corsika.corsika_output.corsikaOutput
+        instance of corsika.corsika_output.corsikaOutput.
+    property_name: string
+        Name of the quantity. Options are: "wavelength", "counts", "density", "time", "altitude",
+        "num_photons".
+    log_y: bool
+        if True, the intensity of the Y axis is given in logarithmic scale.
+
+    Returns
+    -------
+    list
+        List of figures for the given telescopes.
+
+    Raises
+    ------
+    ValueError
+        if `name` is not allowed.
+    """
+
+    x_label = {
+        "wavelength": "Wavelength (nm)",
+        "counts": "Distance to center (m)",
+        "density": "Distance to center (m)",
+        "time": "Time since 1st interaction (ns)",
+        "altitude": "Altitude of emission (km)",
+        "num_photons": "Number of photons per event",
+    }
+    if property_name not in x_label:
+        msg = "results: status must be one of {}".format(x_label)
```

Comment:
Please use f-string everywhere (a decision we made a few months ago and I went through the entire code to make it consistent everywhere).